### PR TITLE
cmd/atlascmd: use env flag for inspect/apply

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: cmd/atlascmd/schema.go [hunk 1]
```diff
@@ -115,101 +122,128 @@ const (
...
 
 	// Schema fmt.
 	schemaCmd.AddCommand(SchemaFmt)
 }
 
+// selectEnv returns the Env from the current project file based on the selected
+// argument. If selected is "", or no project file exists in the current directory
+// a zero-value Env is returned.
+func selectEnv(selected string) (*Env, error) {
+	if _, err := os.Stat(projectFileName); selected == "" || os.IsNotExist(err) {
```
Comment (masseelch) [inline cmd/atlascmd/schema.go:160]: Why hit the disk if selected is zero?

**Review by masseelch — COMMENTED** [review]
I feel like it is more clean to load the env into the flag values before running the command (like in PersistentPreRun e.g.) instead if in each command. 

WDYT?

File: cmd/atlascmd/schema.go [hunk 2]
```diff
@@ -115,101 +122,128 @@ const (
...
+	return LoadEnv(projectFileName, selected)
+}
+
 // CmdInspectRun is the command used when running CLI.
-func CmdInspectRun(cmd *cobra.Command, args []string) {
+func CmdInspectRun(cmd *cobra.Command, _ []string) error {
 	if InspectFlags.Web {
 		schemaCmd.PrintErrln("The Alas UI is not available in this release.")
-		return
+		return errors.New("unsupported")
```
Comment (a8m) [inline cmd/atlascmd/schema.go:170]: Maybe remove the printing and return the message as an error? 

```suggestion
		return errors.New("Atlas UI is not available in this release")
```

File: cmd/atlascmd/schema.go [hunk 3]
```diff
@@ -88,7 +95,7 @@ databases), omit the relevant part from the url, e.g. "mysql://user:pass@localho
 To select specific schemas from the databases, users may use the "--schema" (or "-s" shorthand)
 flag.
 	`,
-		Run: CmdInspectRun,
+		RunE: CmdInspectRun,
```
Comment (a8m) [inline cmd/atlascmd/schema.go:98]: Why did you replace `Run` with `RunE`?

File: cmd/atlascmd/schema.go [hunk 4]
```diff
@@ -115,101 +122,128 @@ const (
...
 	}
-	client, err := sqlclient.Open(cmd.Context(), InspectFlags.URL)
-	cobra.CheckErr(err)
 	defer client.Close()
-	schemas := InspectFlags.Schema
-	activeEnv, err := selectEnv(args)
-	cobra.CheckErr(err)
-	if activeEnv != nil && len(activeEnv.Schemas) > 0 {
-		schemas = activeEnv.Schemas
+	// Select the schemas.
```
Comment (a8m) [inline cmd/atlascmd/schema.go:189]:
```suggestion
```

File: cmd/atlascmd/schema.go [hunk 5]
```diff
@@ -115,101 +122,128 @@ const (
...
+	schemas := activeEnv.Schemas
+	if fl := cmd.Flag(schemaFlag); fl.Changed {
+		schemas = InspectFlags.Schema
 	}
 	if client.URL.Schema != "" {
 		schemas = append(schemas, client.URL.Schema)
 	}
+	// Inspect the realm.
```
Comment (a8m) [inline cmd/atlascmd/schema.go:197]:
```suggestion
```

File: cmd/atlascmd/schema.go [hunk 6]
```diff
@@ -115,101 +122,128 @@ const (
...
 	// Schema fmt.
 	schemaCmd.AddCommand(SchemaFmt)
 }
 
+// selectEnv returns the Env from the current project file based on the selected
+// argument. If selected is "", or no project file exists in the current directory
+// a zero-value Env is returned.
+func selectEnv(selected string) (*Env, error) {
+	if _, err := os.Stat(projectFileName); selected == "" || os.IsNotExist(err) {
+		return &Env{}, nil
```
Comment (a8m) [inline cmd/atlascmd/schema.go:161]: To make the code simpler in CmdInspectRun, I'll suggest doing something like this even though it creates small duplication with CmdApplyRun:

```suggestion
		return &Env{
			URL: InspectFlags.URL,
			Schemas: InspectFlags.Schemas,
		}, nil
```

File: cmd/atlascmd/schema.go (hunk 3 above)
Comment (rotemtam) [reply cmd/atlascmd/schema.go:98]: Honestly just dislike cobra.CheckErr everywhere. Returning an error makes the code seem nicer to me.
Not strongly attached to this

Comment (a8m) [reply cmd/atlascmd/schema.go:98]: I don't mind, I wondered why it was needed in this PR. Let me know when it's ready and I'll review this

**Pull request closed without merging**